wifiutils/src/main/java/com/thanosfisherman/wifiutils/WifiUtils.java

```diff
@@ -342,6 +342,16 @@ public WifiUtilsBuilder patternMatch() {
         return this;
     }
 
+
+    @NonNull
+    @Override
+    public WifiSuccessListener connectWith(@NonNull final String ssid) {
+        mSsid = ssid;
+        mPassword = "doesnt matter"; // FIXME: Cover no password case
```

Comment:
You frequently check for `mPassword` and in some branches code with `mPassword = null` wont enter, so will be unable to process code futher

Reply:
Please initialize `mPassword` with `""` here just to be safe.